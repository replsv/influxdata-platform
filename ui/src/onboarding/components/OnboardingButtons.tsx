// Libraries
import React, {PureComponent, RefObject} from 'react'
import {
  Button,
  ComponentColor,
  ComponentSize,
  ButtonType,
  ComponentStatus,
} from 'src/clockface'

interface Props extends DefaultProps {
  onClickBack: () => void
  nextButtonText: string
  backButtonText: string
  onClickSkip?: () => void
}

interface DefaultProps {
  nextButtonStatus?: ComponentStatus
  showSkip?: boolean
  autoFocusNext?: boolean
  skipButtonText?: string
}

class OnboardingButtons extends PureComponent<Props> {
  public static defaultProps: DefaultProps = {
    nextButtonStatus: ComponentStatus.Default,
    showSkip: false,
    autoFocusNext: true,
    skipButtonText: 'Skip',
  }

  private submitRef: RefObject<Button> = React.createRef()

  public componentDidMount() {
    if (this.props.autoFocusNext) {
      const buttonRef = this.submitRef.current.ref
      if (buttonRef) {
        buttonRef.current.focus()
      }
    }
  }

  public componentDidUpdate() {
    if (this.props.autoFocusNext) {
      const buttonRef = this.submitRef.current.ref
      if (buttonRef) {
        buttonRef.current.focus()
      }
    }
  }

  public render() {
    const {
      backButtonText,
      nextButtonText,
      onClickBack,
      nextButtonStatus,
    } = this.props
    return (
      <div className="wizard--button-container">
        <div className="wizard--button-bar">
          <Button
            color={ComponentColor.Default}
            text={backButtonText}
            size={ComponentSize.Medium}
            onClick={onClickBack}
            data-test="back"
            tabIndex={1}
          />
          <Button
            color={ComponentColor.Primary}
            text={nextButtonText}
            size={ComponentSize.Medium}
            type={ButtonType.Submit}
            data-test="next"
            ref={this.submitRef}
            status={nextButtonStatus}
            tabIndex={0}
          />
        </div>
        {this.skipButton}
      </div>
    )
  }

  private get skipButton(): JSX.Element {
    const {skipButtonText, onClickSkip, showSkip} = this.props
    if (!showSkip) {
      return
    }

    return (
      <div className="wizard--skip-container">
        <Button
          customClass="wizard--skip-button"
          size={ComponentSize.Medium}
          color={ComponentColor.Default}
          text={skipButtonText}
          onClick={onClickSkip}
          data-test="skip"
        />
      </div>
    )
  }
}

export default OnboardingButtons
