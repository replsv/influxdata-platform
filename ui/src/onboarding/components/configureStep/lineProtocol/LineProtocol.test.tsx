// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import {LineProtocol} from 'src/onboarding/components/configureStep/lineProtocol/LineProtocol'
import OnboardingButtons from 'src/onboarding/components/OnboardingButtons'

const setup = (override?) => {
  const props = {
    bucket: 'a',
    org: 'a',
    setLPStatus: jest.fn(),
    ...override,
  }
  const wrapper = shallow(<LineProtocol {...props} />)

  return {wrapper}
}

describe('LineProtocol', () => {
  describe('rendering', () => {
    it('renders!', () => {
      const {wrapper} = setup()
      expect(wrapper.exists()).toBe(true)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('if type is not streaming', () => {
    it('renders back, next, and skip buttons with correct text', () => {
      const {wrapper} = setup()
      const buttons = wrapper.find(OnboardingButtons)

      expect(buttons.prop('backButtonText')).toBe(
        'Back to Select Data Source Type'
      )
      expect(buttons.prop('nextButtonText')).toBe('Continue to Verify')
      expect(buttons.prop('skipButtonText')).toBe('Skip Config')
    })
  })
})
