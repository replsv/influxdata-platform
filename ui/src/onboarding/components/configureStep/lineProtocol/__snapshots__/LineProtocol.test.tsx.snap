// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineProtocol rendering renders! 1`] = `
<Form
  onSubmit={[Function]}
>
  <div
    className="wizard-step--scroll-area"
  >
    <FancyScrollbar
      autoHeight={false}
      autoHide={false}
      maxHeight={null}
      setScrollTop={[Function]}
      style={Object {}}
    >
      <div
        className="wizard-step--scroll-content"
      >
        <h3
          className="wizard-step--title"
        >
          Add Data via Line Protocol
        </h3>
        <h5
          className="wizard-step--sub-title"
        >
          Need help writing InfluxDB Line Protocol? See Documentation
        </h5>
        <Connect(LineProtocolTabs)
          bucket="a"
          org="a"
          tabs={
            Array [
              "Upload File",
              "Enter Manually",
              "Enter URL",
            ]
          }
        />
      </div>
    </FancyScrollbar>
  </div>
  <OnboardingButtons
    autoFocusNext={true}
    backButtonText="Back to Select Data Source Type"
    nextButtonStatus="default"
    nextButtonText="Continue to Verify"
    showSkip={true}
    skipButtonText="Skip Config"
  />
</Form>
`;
